nautobot/extras/models/models.py

```diff
@@ -196,7 +197,24 @@ def get_config_context(self):
             config_context_data = ConfigContext.objects.get_for_object(self).values_list("data", flat=True)
         else:
             config_context_data = self.config_context_data or []
+            # Device and VirtualMachine's Location has its own ConfigContext and its parent Locations' ConfigContext, if any, should
+            # also be applied. However, since moving from mptt to django-tree-queries https://github.com/nautobot/nautobot/issues/510,
+            # we lost the ability to query the ancestors for a particular tree node for subquery https://github.com/matthiask/django-tree-queries/issues/54.
+            # So instead of constructing the location related query in ConfigContextModelQueryset._get_config_context_filters(), which is complicated across databases
+            # We append the missing parent location query here as a patch.
+            if self._meta.model_name == "device":
+                location_config_context_queryset = ConfigContext.objects.filter(
+                    Q(locations__in=self.location.ancestors(include_self=True))
+                ).distinct()
+            else:
+                if self.cluster:
+                    location_config_context_queryset = ConfigContext.objects.filter(
+                        Q(locations__in=self.cluster.location.ancestors(include_self=True))
```

Comment:
`Location` is not required for `Cluster` so this could raise an `AttributeError`